src/BuildScriptGenerator/Helpers/ParserHelper.cs

```diff
@@ -29,18 +29,18 @@ public static TomlTable ParseTomlFile(ISourceRepo sourceRepo, string filePath)
         }
 
         /// <summary>
-        /// Parse a .yaml file into a YamlMappingNode from the YamlDotNet library.
+        /// Parse a .yaml file into a YamlNode from the YamlDotNet library.
         /// See https://github.com/aaubry/YamlDotNet for more information.
         /// </summary>
         /// <param name="sourceRepo">Source repo for the application.</param>
         /// <param name="filePath">The path to the .yaml file.</param>
-        /// <returns>A YamlMappingNode object containing information about the .yaml file.</returns>
-        public static YamlMappingNode ParseYamlFile(ISourceRepo sourceRepo, string filePath)
+        /// <returns>A YamlNode object containing information about the .yaml file.</returns>
+        public static YamlNode ParseYamlFile(ISourceRepo sourceRepo, string filePath)
```

Comment:
Ah can you check if this helper is indeed used in BuildScriptGenerator since i copied it to Detector too. If it is not used we can delete this file.

Reply:
I found this is no longer used so deleted it, thanks for reminding, and also removed projectFileProviders related from generator since we already have that in detector.